pkg/availability/multisigcollector/multisigcollector.go

```diff
@@ -48,3 +52,40 @@ func NewModule(mc *ModuleConfig, params *ModuleParams, nodeID t.NodeID) (modules
 
 	return m, nil
 }
+
+func NewReconfigurableModule(mc *ModuleConfig, nodeID t.NodeID, logger logging.Logger) modules.PassiveModule {
+	return factorymodule.New(
+		mc.Self,
+		factorymodule.DefaultParams(
+
+			// This function will be called whenever the factory module
+			// is asked to create a new instance of the multisig collector.
+			func(mscID t.ModuleID, params *factorymodulepb.GeneratorParams) (modules.PassiveModule, error) {
+
+				// Extract the IDs of the nodes in the membership associated with this instance
+				m := params.Type.(*factorymodulepb.GeneratorParams_MultisigCollector).MultisigCollector.Membership
+				mscNodeIDs := maputil.GetSortedKeys(t.Membership(m))
+
+				// Crate a copy of basic module config with an adapted ID for the submodule.
+				submc := *mc
+				submc.Self = mscID
+
+				// Create a new instance of the multisig collector.
+				multisigCollector, err := NewModule(
+					&submc,
+					&ModuleParams{
+						InstanceUID: []byte(mscID),
+						AllNodes:    mscNodeIDs,
+						F:           (len(mscNodeIDs) - 1) / 2,
```

Comment:
Works for now, but a bit weird :)
It is likely higher than what other modules will use (likely, (N-1)/3 or lower)

Reply:
I copied that from your demo :-)
But if it works for the MSC, let's keep it, it's a MSC-specific configuration parameter anyway.

Reply:
> I copied that from your demo :-)
It makes sense for the demo because there are no other modules :)

> But if it works for the MSC, let's keep it
It's just that it is a bit wasteful (the certificate will contain roughly 50% more signatures), but yes, not a big deal.
Maybe leave a TODO though?

Feel free to mark this conversation as resolved with or without TODO :)

Reply:
Yes I see. If other modules have a lower fault tolerance, there is no point waiting for too many signatures here.
I'll add a TODO and merge.